client/apis/volumesnapshot/v1/types.go

```diff
@@ -328,6 +328,12 @@ type VolumeSnapshotContentSpec struct {
 	// This field is immutable after creation.
 	// Required.
 	Source VolumeSnapshotContentSource `json:"source" protobuf:"bytes,5,opt,name=source"`
+
+	// SourceVolumeMode is the mode of the volume whose snapshot is taken.
+	// Can be either “Filesystem” or “Block”.
+	// If not specified, it indicates the source volume's mode is unknown.
+	// +optional
+	SourceVolumeMode *SourceVolumeMode `json:"sourceVolumeMode" protobuf:"bytes,6,opt,name=sourceVolumeMode"`
```

Comment:
I'm thinking this field should be immutable. Otherwise it is possible for someone to modify this and make it the same as the target volume mode.
See example here: https://github.com/kubernetes-csi/external-snapshotter/blob/v5.0.1/client/apis/volumesnapshot/v1/types.go#L109

I'm fine to get this merged first. Then you can submit a follow up PR to make this field immutable and add logic in the validation webhook: https://github.com/kubernetes-csi/external-snapshotter/blob/v5.0.1/pkg/validation-webhook/snapshot.go#L266

Reply:
Yes that makes sense to me too. Will update that in a follow up PR